analyzers/dataframe/MCParticle.cc

```diff
@@ -679,6 +679,97 @@ ROOT::VecOps::RVec<float> MCParticle::AngleBetweenTwoMCParticles( ROOT::VecOps::
 
 }
 
+// --------------------------------------------------------------------------------------------------
+
+int MCParticle::get_lepton_origin( edm4hep::MCParticleData p, ROOT::VecOps::RVec<edm4hep::MCParticleData> in, ROOT::VecOps::RVec<int> ind)  {
+
+ // std::cout  << std::endl << " enter in MCParticle::get_lepton_origin  PDG = " << p.PDG << std::endl;
+
+ int pdg = std::abs( p.PDG ) ;
+ if ( pdg != 11 && pdg != 13 && pdg  != 15 ) return -1; 
+
+ int result  = 0;
+
+ // std::cout << " p.parents_begin p.parents_end " << p.parents_begin <<  " "  << p.parents_end << std::endl;
+    for (unsigned j = p.parents_begin; j != p.parents_end; ++j) {
+      int index = ind.at(j);
```

Comment:
again, talking to RDF folks, we should avoid using ```operator.at()``` but rather ```operator[]``` which is much faster